synapse/lib/snap.py

```diff
@@ -1305,6 +1305,11 @@ async def _applyNodeEdits(self, edits, meta, nodecache=None):
             mesg = 'The snapshot is in read-only mode.'
             raise s_exc.IsReadOnly(mesg=mesg)
 
+        useriden = meta.get('user')
+        if useriden is None:
+            mesg = 'meta is missing user key. Cannot process edits.'
+            raise s_exc.BadArg(mesg=mesg, name='user')
```

Comment:
Should we fall back to the root user instead of raising here?

Reply:
Yeah it seems unlikely to happen, but in the case where malformed edits are coming in it seems like it would be fine for `useriden` to be None. Mostly just looking to not break triggers if that is somehow happening somewhere.